Tella/Data/Database/TellaData.swift

```diff
@@ -125,6 +125,23 @@ class TellaData : ObservableObject {
         return database.getReport(reportId: reportId)
     }
     
+    func getResources() -> [DownloadedResource] {
+        return database.getDownloadedResources()
+    }
+    
+    func addResource(resource: Resource, serverId: Int) throws -> String {
+        let result = database.addDownloadedResource(resource: resource, serverId: serverId)
+        
+        switch result {
+        case .success(let resourceId):
+            return resourceId
+        case .failure(let error):
+            throw error
+        }
+    }
+    func deleteDownloadedResource(resourceId: String) -> Void {
+        let _ = database.deleteDownloadedResource(resourceId: resourceId)
```

Comment:
It's better to return the bool result  `Result<Bool,Error>` to treat it in case success or failure